client/html/src/Client/Html/Catalog/Product/Standard.php

```diff
@@ -0,0 +1,468 @@
+		 *
+		 * @param array List of decorator names
+		 * @since 2019.06
+		 * @category Developer
+		 * @see client/html/common/decorators/default
+		 * @see client/html/catalog/product/decorators/excludes
+		 * @see client/html/catalog/product/decorators/global
+		 */
+
+		return $this->createSubClient( 'catalog/product/' . $type, $name );
+	}
+
+
+	/**
+	 * Processes the input, e.g. store given values.
+	 * A view must be available and this method doesn't generate any output
+	 * besides setting view variables.
+	 */
+	public function process()
+	{
+		$context = $this->getContext();
+		$view = $this->getView();
+
+		try {
+			parent::process();
+		} catch (\Aimeos\Client\Html\Exception $e) {
+			$error = array($context->getI18n()->dt( 'client', $e->getMessage() ));
+			$view->productErrorList = $view->get( 'productErrorList', [] ) + $error;
+		} catch (\Aimeos\Controller\Frontend\Exception $e) {
+			$error = array($context->getI18n()->dt( 'controller/frontend', $e->getMessage() ));
+			$view->productErrorList = $view->get( 'productErrorList', [] ) + $error;
+		} catch (\Aimeos\MShop\Exception $e) {
+			$error = array($context->getI18n()->dt( 'mshop', $e->getMessage() ));
+			$view->productErrorList = $view->get( 'productErrorList', [] ) + $error;
+		} catch (\Exception $e) {
+			$error = array($context->getI18n()->dt( 'client', 'A non-recoverable error occured' ));
+			$view->productErrorList = $view->get( 'productErrorList', [] ) + $error;
+			$this->logException( $e );
+		}
+	}
+
+
+	/**
+	 * Returns the list of sub-client names configured for the client.
+	 *
+	 * @return array List of HTML client names
+	 */
+	protected function getSubClientNames()
+	{
+		return $this->getContext()->getConfig()->get( $this->subPartPath, $this->subPartNames );
+	}
+
+
+	/**
+	 * Modifies the cached body content to replace content based on sessions or cookies.
+	 *
+	 * @param string $content Cached content
+	 * @param string $uid Unique identifier for the output if the content is placed more than once on the same page
+	 * @return string Modified body content
+	 */
+	public function modifyBody( $content, $uid )
+	{
+		$content = parent::modifyBody( $content, $uid );
+
+		return $this->replaceSection( $content, $this->getView()->csrf()->formfield(), 'catalog.lists.items.csrf' );
+	}
+
+	/**
+	 * Sets the necessary parameter values in the view.
+	 *
+	 * @param \Aimeos\MW\View\Iface $view The view object which generates the HTML output
+	 * @param array &$tags Result array for the list of tags that are associated to the output
+	 * @param string|null &$expire Result variable for the expiration date of the output (null for no expiry)
+	 * @return \Aimeos\MW\View\Iface Modified view object
+	 */
+	public function addData( \Aimeos\MW\View\Iface $view, array &$tags = [], &$expire = null )
+	{
+		$context = $this->getContext();
+		$config = $context->getConfig();
+
+		$productItems = [];
+
+		/** client/html/catalog/domains
```

Comment:
You don't need to repeat the documentation that's already available in another file

Reply:
Removed